wal/wal.go

```diff
@@ -834,3 +836,34 @@ func (r *segmentBufReader) Read(b []byte) (n int, err error) {
 	r.buf.Reset(r.segs[r.cur])
 	return n, nil
 }
+
+func dirSize(dir string) (size int64, err error) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return 0, err
+	}
+	for _, fi := range files {
```

Comment:
We can simplify iterating through files/directories using `filepath.Walk` as used here https://github.com/prometheus/tsdb/blob/d5b3f0704379a9eaca33b711aa0097f001817fc2/testutil/directory.go#L136-L148

Reply:
That makes sense; won't have manually recurse through the subdirectories then.

Do you think it also makes sense to check if the file we are currently reading is directly under `wal/` or `wal/checkpoint.XXXX/` to avoid considering files that are not related to the WAL?

Reply:
We would need to consider the checkpoint data too.

Reply:
Taking a look at the files the walker considers, it does consider the checkpoint data too.